### ISSUE TITLE
Vue | Page issue: <v-ons-page> and <v-ons-bottom-toolbar> sliding under issue when custom class defined on <v-ons-page>

### ISSUE THREAD

**Issue by subdigit** [OnsenUI/OnsenUI#2459]
__Environment__

  
```
[Core]
  onsenui latest
  
[Framework]
  vue 2.4.1

[Framework binding]
 vue-onsenui latest

[Platform]
  Desktop - OS X 10.12.5 64-bit

[Browser]
  Desktop - Chrome 69.0.3464.0 on OS X 10.12.5 64-bit
```

__Encountered problem__
  When a &lt;v-ons-bottom-toobar&gt; is present and there is enough content in the "content" section of a &lt;v-ons-page&gt; to trigger the scrollbar, the contents of &lt;v-ons-page&gt; slides under a &lt;v-ons-bottom-toolbar&gt; if a class is defined on &lt;v-ons-page&gt;.

__How to reproduce__
  Have enough content in a &lt;v-ons-page&gt; to toggle the scrollbar and have a &lt;v-ons-bottom-toolbar&gt; present.  Add a class="anything you want" to &lt;v-ons-page&gt;.  Scrolling slides under the &lt;v-ons-bottom-toolbar&gt;.  The class "page-with-bottom-toolbar" never successfully gets added to &lt;v-ons-page&gt; (even though other classes are added successfully).

  
  [__Demo link__](https://onsen.io/playground/?issue)

  

- __HTML__

```html
<template id="main">
  <v-ons-page class="random classes">
    <v-ons-toolbar>
      <div class="center">{{ title }}</div>
      <div class="right">
        <v-ons-toolbar-button>
          <v-ons-icon icon="ion-navicon, material: md-menu"></v-ons-icon>
        </v-ons-toolbar-button>
      </div>
    </v-ons-toolbar>

    <p style="text-align: center">
      <v-ons-button @click="$ons.notification.alert('Hello World!')">
        Click me!
      </v-ons-button>
    </p>

    <p style="text-align: center">
      <v-ons-button @click="$ons.notification.alert('Hello World!')">
        Click me!
      </v-ons-button>
    </p>

    <p style="text-align: center">
      <v-ons-button @click="$ons.notification.alert('Hello World!')">
        Click me!
      </v-ons-button>
    </p>

    <p style="text-align: center">
      <v-ons-button @click="$ons.notification.alert('Hello World!')">
        Click me!
      </v-ons-button>
    </p>

    <p style="text-align: center">
      <v-ons-button @click="$ons.notification.alert('Hello World!')">
        Click me!
      </v-ons-button>
    </p>

    <p style="text-align: center">
      <v-ons-button @click="$ons.notification.alert('Hello World!')">
        Click me!
      </v-ons-button>
    </p>

    <p style="text-align: center">
      <v-ons-button @click="$ons.notification.alert('Hello World!')">
        Fourth from last
      </v-ons-button>
    </p>

    <p style="text-align: center">
      <v-ons-button @click="$ons.notification.alert('Hello World!')">
        Third from last
      </v-ons-button>
    </p>

    <p style="text-align: center">
      <v-ons-button @click="$ons.notification.alert('Hello World!')">
        Second to last
      </v-ons-button>
    </p>

    <p style="text-align: center">
      <v-ons-button @click="$ons.notification.alert('Hello World!')">
        Last One
      </v-ons-button>
    </p>

    <v-ons-bottom-toolbar>
      The buttons will slide under me.
    </v-ons-bottom-toolbar>
  </v-ons-page>
</template>

<div id="app"></div>
```

- __JS__

```javascript
new Vue({
  el: '#app',
  template: '#main',
  data() {
    return {
      title: 'My app'
    };
  }
});
```

**Comment by puku0x** [comment]
I made a pen for reproduction because it seems the demo link is broken.

This is for `vue-onsenui`.
https://codepen.io/puku0x/pen/dKgNLo

It works without Vue framework binding, though 🤔
https://codepen.io/puku0x/pen/mKzWBB

**Comment by puku0x** [comment]
I found `page-with-bottom-toolbar` class was missing when using `v-ons-page` with custom class.

Current workaround is to add `page-with-bottom-toolbar`.
```html
<v-ons-page class="random classes page-with-bottom-toolbar">
```

**Comment by subdigit** [comment]
Thanks for the demo update!  I did notice it worked when I manually added the class, so I have a temporary workaround.

So it sounds like a Vue and some kind of class binding issue.  The `page` class gets added properly, but just not the `page-with-bottom-toolbar`

**Comment by asialgearoid** [comment]
I've managed to figure out what's causing this, but I'm not sure how to fix it.

The steps go something like below. It's worth noting that `v-ons-page` is basically a wrapper for `ons-page`, and most of the operations are directly on `ons-page`. 
1. `v-ons-page` creates `ons-page` - `<ons-page class="page">`
2. `ons-bottom-toolbar` adds the `page-with-bottom-toolbar` class - `<ons-page class="page page-with-bottom-toolbar">`
3.  Vue finds the custom classes, and sets them on `ons-page`, overwriting the existing ones - `<ons-page class="random classes">`
4. Onsen UI detects this class change, and re-adds the page's default classes - `<ons-page class="page random classes">`

The problem here is step 4. There's nothing we can do to prevent Vue from overwriting the class, but we can restore the previous classes, which is what step 4 is supposed to do. What it actually restores is the default classes - `page-with-bottom-toolbar` is not one of these, because it is added to the page by the toolbar. I'm not sure this is the correct approach. I think the page should add its own class if it has a toolbar child.

I will need to think about this a bit more, so I suggest using @puku0x's workaround for now.

**Comment by asialgearoid** [comment]
I've tested with this for a while since my last comment, and I think that step 2 should be handled by `ons-page`, not `ons-bottom-toolbar`. The one problem is that this doesn't handle the case where the toolbar is added after the page is loaded, but I'm pretty sure there would be other unrelated issue in this case anyway. I have created a PR, which hopefully will get merged soon.

**Comment by asialgearoid** [comment]
@subdigit This fix is in `onsenui@2.10.2`. I'm not sure if I mentioned previously, but the issue was actually in the core, not the Vue bindings. So, just update to the latest version of Onsen UI and you'll be good to go.

**Comment by subdigit** [comment]
Thanks!  Once we release with the current version (2.9.2) I'll update the core and remove the hack fix.
